functests/temp/ptp-tmp.go

```diff
@@ -0,0 +1,492 @@
+}
+
+func getPtpPodOnNode(nodeName string) (v1core.Pod, error) {
+	runningPod, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprint("Error to get list of pods by label: app=linuxptp-daemon"))
+	Expect(len(runningPod.Items)).To(BeNumerically(">", 0), fmt.Sprint("PTP pods are  not deployed on cluster"))
+	for _, pod := range runningPod.Items {
+
+		if pod.Spec.NodeName == nodeName {
+			return pod, nil
+		}
+	}
+	return v1core.Pod{}, fmt.Errorf("Pod not found")
+}
+
+func getMasterSlaveAttachedInterfaces(pod v1core.Pod) []string {
+	var IntList []string
+	Eventually(func() error {
+		buf, err := pods.ExecCommand(client.Client, pod, []string{"ls", "/sys/class/net/"})
+		if err != nil {
+			return err
+		}
+
+		IntList = strings.Split(strings.Join(strings.Fields(buf.String()), " "), " ")
+		if len(IntList) == 0 {
+			return fmt.Errorf("No interface detected")
+		}
+
+		return nil
+	}, 3*time.Minute, 2*time.Second).Should(BeNil())
+
+	return IntList
+}
+
+func getPtpMasterSlaveAttachedInterfaces(pod v1core.Pod) []string {
+	var ptpSupportedInterfaces []string
+	var buf bytes.Buffer
+
+	intList := getMasterSlaveAttachedInterfaces(pod)
+	for _, interf := range intList {
+
+		Eventually(func() error {
+			var err error
+			buf, err = pods.ExecCommand(client.Client, pod, []string{"ethtool", "-T", interf})
+			if err != nil {
+				return err
+			}
+			return nil
+		}, 2*time.Minute, 1*time.Second).Should(BeNil())
+
+		if isPTPEnabled(&buf) {
+			ptpSupportedInterfaces = append(ptpSupportedInterfaces, interf)
+		}
+	}
+	return ptpSupportedInterfaces
+}
+
+func replaceTestPod(pod v1core.Pod, timeout time.Duration) (v1core.Pod, error) {
+	var newPod v1core.Pod
+
+	err := client.Client.Pods(ptpLinuxDaemonNamespace).Delete(pod.Name, &metav1.DeleteOptions{
+		GracePeriodSeconds: pointer.Int64Ptr(0)})
+	Expect(err).NotTo(HaveOccurred())
+
+	Eventually(func() error {
+		newPod, err = getPtpPodOnNode(pod.Spec.NodeName)
+
+		if err == nil && newPod.Name != pod.Name && newPod.Status.Phase == "Running" {
+			return nil
+		}
+
+		return fmt.Errorf("Can not replace PTP pod")
+	}, timeout, 1*time.Second).Should(BeNil())
+
+	return newPod, nil
+}
+
+func getNonPtpMasterSlaveAttachedInterfaces(pod v1core.Pod) []string {
+	var ptpSupportedInterfaces []string
+	var buf bytes.Buffer
+
+	intList := getMasterSlaveAttachedInterfaces(pod)
+	for _, interf := range intList {
+
+		Eventually(func() error {
+			var err error
+			buf, err = pods.ExecCommand(client.Client, pod, []string{"ethtool", "-T", interf})
+			if err != nil {
+				return err
+			}
+			return nil
+		}, 2*time.Minute, 1*time.Second).Should(BeNil())
+		if isPTPEnabled(&buf) == false {
+			ptpSupportedInterfaces = append(ptpSupportedInterfaces, interf)
+		}
+		time.Sleep(time.Second)
+	}
+	return ptpSupportedInterfaces
+}
```

Comment:
why is this temp?

Reply:
I miss that here is a PR to fix it

https://github.com/openshift-kni/cnf-features-deploy/pull/174

Thanks for finding it :)

Reply:
try to avoid 
git add -a 
next time ;-)